staging/src/k8s.io/apiserver/pkg/server/filters/waitgroup.go

```diff
@@ -38,7 +43,13 @@ func WithWaitGroup(handler http.Handler, longRunning apirequest.LongRunningReque
 
 		if !longRunning(req, requestInfo) {
 			if err := wg.Add(1); err != nil {
-				http.Error(w, "apiserver is shutting down.", http.StatusInternalServerError)
+				// When apiserver is shutting down, signal clients to retry
+				w.Header().Add("Retry-After", "1")
```

Comment:
Add a comment explaining that the expectation is that with a load-balancer, you'll  hit a different server, so a tight retry is good for client responsiveness.